File: solutions/typescript/0009.ts
function isPalindrome(x: number): boolean {
    if (x < 0) return false;
    let reversed = 0;
    let original = x;
    while (x > 0) {
        reversed = reversed * 10 + x % 10;
        x = Math.floor(x / 10);
    }
    return reversed === original;
};

File: solutions/java/0009.java
public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        int reversed = 0;
        int original = x;
        while (x > 0) {
            reversed = reversed * 10 + x % 10;
            x = x / 10;
        }
        return reversed == original;
    }
}